models/db/engine.go

```diff
@@ -293,3 +293,26 @@ func GetMaxID(beanOrTableName interface{}) (maxID int64, err error) {
 	_, err = x.Select("MAX(id)").Table(beanOrTableName).Get(&maxID)
 	return
 }
+
+// QuoteString quotes the string for SQL. In most cases, SQL should use the ORM builder, not this one.
+// This function is NOT recommended, it could ONLY be used in rare cases when there is no other choice.
+// It panics if the database is not supported to avoid corrupted data
+// At the moment it follows the behavior of convertString and convertStringSingleQuote in XORM, and is covered by tests.
+func QuoteString(ctx context.Context, s string) string {
+	e, ok := GetEngine(ctx).(*xorm.Engine)
+	if !ok {
+		panic("can not get the real database engine")
+	}
+
+	switch e.Dialect().URI().DBType {
```

Comment:
```suggestion
	// The NUL char doesn't need to be considered because there are undefined behaviors across databases, as long as the NUL char doesn't do harm, let it go.
	switch e.Dialect().URI().DBType {
```